### ISSUE TITLE
stat(2) fails for existing entries

### ISSUE THREAD

**Issue by sandrain** [LLNL/UnifyCR#166]
UnifyCR maintains file metadata records in the local fid data structure (`unifycr_filelist` and `unifycr_filemetas`) and also in the global metadata db. The `fid` data structure is populated from the global metadata db, only when the `open(2)` system is invoked.  Therefore, when an application execute the `stat(2)` system call before opening a file that only exists in the global metadata db, it will fail with `ENOENT`.

In specific, the current `stat(2)` implementation does not check the global metadata database, but
only calls the `unifycr_get_fid_from_path` function to check whether the given file locally exists or not:

```
/* given a path, return the file id */
inline int unifycr_get_fid_from_path(const char *path)
{
    int i = 0;
    while (i < unifycr_max_files) {
        if (unifycr_filelist[i].in_use &&
            strcmp((void *)&unifycr_filelist[i].filename, path) == 0) {
            DEBUG("File found: unifycr_filelist[%d].filename = %s\n",
                  i, (char *)&unifycr_filelist[i].filename);
            return i;
        }
        i++;
    }

    /* couldn't find specified path */
    return -1;
}
```

**Comment by sandrain** [comment]
Addressed by #167. This can be closed If the PR is merged.

**Comment by sandrain** [comment]
This has been addressed by #167.
